### ISSUE TITLE
vscode `editor.tabCompletion` and `editor.suggest.preview` are not working as expected with rust-analyzer

### ISSUE THREAD

**Issue by GilShoshan94** [microsoft/vscode#129443]
Does this issue occur when all extensions are disabled?: Not relevant I think as I need the Rust Analyzer extension.

- VS Code Version: 1.58.2
- OS Version: Windows 10 (64 bits)

Hi, I already open this issue in the rust-analyzer project as it seemed more probable an issue on their side but after reply of two of their contributors (@Veykril and @flodiebold) suggest I open the issue here in vscode as well.

Here is the original issue: rust-analyzer/rust-analyzer#9698

To recap, the settings `editor.tabCompletion` and `editor.suggest.preview` do not work at all.
Those settings have no impact at all.

It sounds like they should work out of the box with the Rust Analyzer (it's a language server for Rust).
But they don't.

Can someone from the vscode team help us with this please ?
Could you provide developer docs on these features, that is how to support these in a client please ?

Thank you in advance.
Gil

**Comment by hediet** [comment]
Can you post a screenshot/video demonstrating the issue?
What is the content of the file, where is your cursor?

**Comment by GilShoshan94** [comment]
Hi @hediet,

I provided more detailed info (`keybindings.json`, `settings.json` and list of installed extension) in the first issue I opened at [rust-analyzer/rust-analyzer#9698](https://github.com/rust-analyzer/rust-analyzer/issues/9698).

You can take an empty Rust file (for example `main.rs`) and start writing standard function/method/macro, such as `print!(...)` and `println(...)`.

The suggestion list works fine as you can see:
![image](https://user-images.githubusercontent.com/29143941/129563695-baf1354c-25c3-495f-8289-069faf73da93.png)

But the `editor.suggest.preview` doesn't work, and the more important problem, once I press `TAB`, the first selected suggestion get inserted, as such:
![image](https://user-images.githubusercontent.com/29143941/129563985-be683d61-99e4-49c4-82f7-9143e02c8c0f.png)

But now when I press `TAB` again, instead of cycling over the next suggestion (which should be `println(...)`), I get one space inserted as shown:
![image](https://user-images.githubusercontent.com/29143941/129564243-f22a5a01-7385-40a7-83f9-4ecb2db1fc58.png)
 
And with `SHIFT + TAB`, instead of reverse cycling, I get a deindent:
![image](https://user-images.githubusercontent.com/29143941/129564429-d46ae618-bfad-450d-9bfe-86c74dfb5b1f.png)

**Comment by GilShoshan94** [comment]
The cursor is in the editor, in the code.

**Comment by hediet** [comment]
Fixed 

![image](https://user-images.githubusercontent.com/2931520/130754645-253ae077-6f59-4622-a77f-73f0686cb184.png)

**Comment by GilShoshan94** [comment]
@hediet 
Thank you.
Will it be available in the next update ?

**Comment by hediet** [comment]
It will be available in tomorrows insider build!

**Comment by GilShoshan94** [comment]
Great, thank you !
